### PR TITLE
Disallow subprojects have incompatible deps

### PR DESCRIPTION
**Pull request by NickeZ**
For now it seems that meson cannot handle the case when two subprojects have incompatible dependencies. This will make it fail.

### CODE REVIEW

**Comment by jpakkane** [comment]
Please squash the fixups and rebase so CI gets run on this again. We want to make sure that the cygwin failure is unrelated.

**Comment by jpakkane** [comment]
ping

**Comment by NickeZ** [comment]
If this passes CI, it is ready to be merged.

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -2131,6 +2131,8 @@ def _find_cached_dep(self, name, kwargs):
         # Check if we've already searched for and found this dep
         if identifier in self.coredata.deps:
             cached_dep = self.coredata.deps[identifier]
+            mlog.log('Cached dependency', mlog.bold(name),
+                    'found:', mlog.green('YES'))
```
Comment (ignatenkobrain) [inline mesonbuild/interpreter.py]: __[Flake8]__

continuation line under-indented for visual indent

**Pull request merged**